File: 3442 maximum-difference-between-even-and-odd-frequency-i/solution.java
class Solution {
    public int maxDifference(String s) {
        int[] arr = new int[26];
        for(char c:s.toCharArray()) {
            arr[c-'a']++;
        }
        int maxO = -1,minE = 101;
        for(int i:arr) {
            if(i==0) continue;
            if((i & 1) == 0) {
                minE = Math.min(minE,i);
            } else if(i>maxO) {
                maxO = i;
            }
        }
        return maxO - minE;
    }
}

File: 3442 maximum-difference-between-even-and-odd-frequency-i/solution.cpp
class Solution {
public:
    int maxDifference(string s) {
        vector<int> f(26);
        for(char c : s) {
            f[c-'a']++;
        }
        int minE = 101,maxO = -1;
        for(int i : f) {
            if(i==0) continue;
            if((i & 1) ==0) {
                minE = min(minE,i);
            } else if(i > maxO) {
                maxO = i;
            }
        }
        return maxO - minE;
    }
};
